Inventory.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace InventoryManagement
+{
+    public class Inventory
+    {
+
+        List<Product> products = new List<Product>();
+
+        //Define Constructor 
+        public Inventory() { }
+
+        //Add new Product to the List 
+
+        public void AddProduct(Product product)
+        {
+            products.Add(product);
+        }
+
+        //Delete Product from the List 
+
+        public void RemoveProduct(Product product)
+        {
+            products.Remove(product);
+        }
+
+        //Display All Products in the List 
+        public void ViewAllProducts()
+        {
+            foreach (var product in products)
+            {
+                Console.WriteLine("Name={0}\tQuantity={1}\tPrice={2}$", product.Name, product.Quantity, product.Price);
+            }
+        }
+
+        //Search for a Product in the List using It's Name
+        public Product SearchProduct(String name)
+        {
+            Product searchedProduct = null;
+            foreach (var product in products)
+            {
+                if (product.Name.ToLower() == name.ToLower()) searchedProduct = product;
+            }
+            return searchedProduct;
```

Comment:
```suggestion
            foreach (var product in products)
            {
                if (product.Name.ToLower() == name.ToLower()) return product;
            }
            return null;
```

There's an advantage of doing this other than code simplicity, that it makes it clear how the method works:
1. return the product if it matches the name,
2. otherwise return `null`.

Reply:
Yes, certainly. This approach seems clearer and more efficient, especially when it comes to catching the product without the need to iterate through the remaining products.

- I updated the code using this